controllers/octaviaapi_controller.go

```diff
@@ -672,11 +829,23 @@ func (r *OctaviaAPIReconciler) generateServiceConfigMaps(
 
 	cmLabels := labels.GetLabels(instance, labels.GetGroupLabel(octavia.ServiceName), map[string]string{})
 
+	db, err := mariadbv1.GetDatabaseByName(ctx, h, octavia.DatabaseName)
+	if err != nil {
+		return err
+	}
+	var tlsCfg *tls.Service
+	if instance.Spec.TLS.CaBundleSecretName != "" {
+		tlsCfg = &tls.Service{}
+	}
+
 	// customData hold any customization for the service.
 	// custom.conf is going to /etc/<service>/<service>.conf.d
 	// all other files get placed into /etc/<service> to allow overwrite of e.g. logging.conf or policy.json
 	// TODO: make sure custom.conf can not be overwritten
-	customData := map[string]string{common.CustomServiceConfigFileName: instance.Spec.CustomServiceConfig}
+	customData := map[string]string{
+		common.CustomServiceConfigFileName: instance.Spec.CustomServiceConfig,
+		"my.cnf":                           db.GetDatabaseClientConfig(tlsCfg), //(mschuppert) for now just get the default my.cnf
+	}
```

Comment:
this controller renders the `httpd.conf` template and the templateParameters misses the VHosts config, therefore the controller fails with:

```
2024-03-15T14:56:49.355Z	ERROR	Controllers.OctaviaAPI	unable to process config map	{"controller": "octaviaapi", "controllerGroup": "octavia.openstack.org", "controllerKind": "OctaviaAPI", "OctaviaAPI": {"name":"octavia-api","namespace":"octavia-kuttl-tests"}, "namespace": "octavia-kuttl-tests", "name": "octavia-api", "reconcileID": "94894f92-81c8-4605-aef8-54b4ceb17197", "error": "template: tmp:26:27: executing \"tmp\" at <.VHosts>: map has no entry for key \"VHosts\""}
github.com/openstack-k8s-operators/octavia-operator/controllers.(*OctaviaAPIReconciler).generateServiceConfigMaps
	/remote-source/controllers/octaviaapi_controller.go:958
github.com/openstack-k8s-operators/octavia-operator/controllers.(*OctaviaAPIReconciler).reconcileNormal
	/remote-source/controllers/octaviaapi_controller.go:647
github.com/openstack-k8s-operators/octavia-operator/controllers.(*OctaviaAPIReconciler).Reconcile
	/remote-source/controllers/octaviaapi_controller.go:184
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:119
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:316
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:266
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.16.5/pkg/internal/controller/controller.go:227
```

Reply:
I put the vhost config to the octavia_controller, and totally overlooked this, thanks!

Reply:
Should be resolved, but I'll wait after a successful kuttl run, which is now failing because of pull image error.

Reply:
Locally passed